src/main/java/org/junitpioneer/jupiter/IssueTestCase.java

```diff
@@ -61,23 +65,36 @@ public Status result() {
 		return result;
 	}
 
+	/**
+	 * Returns the elapsed time since the start of test methods' execution in milliseconds.
+	 *
+	 * @return The elapsed time in ms.
+	 */
+	public Optional<Long> elapsedTime() {
+		return Optional.ofNullable(elapsedTime);
+	}
+
 	@Override
 	public boolean equals(Object o) {
 		if (this == o)
 			return true;
 		if (!(o instanceof IssueTestCase))
 			return false;
 		IssueTestCase that = (IssueTestCase) o;
-		return testId.equals(that.testId) && result == that.result;
+		return testId.equals(that.testId) && result == that.result && Objects.equals(elapsedTime, that.elapsedTime);
 	}
 
 	@Override
 	public int hashCode() {
-		return Objects.hash(testId, result);
+		return Objects.hash(testId, result, elapsedTime);
 	}
 
 	@Override
 	public String toString() {
+		if (Objects.nonNull(elapsedTime)) {
+			return "IssueTestCase{" + "uniqueName='" + testId + '\'' + ", result='" + result + '\'' + ", elapsedTime='"
```

Comment:
Why double all the code and not just add the additional time information when it's present?